### ISSUE TITLE
`expr.str.extract_matches()` to access `regex::Match` struct impl results  (e.g. `range`, `start`, `end`, etc)

### ISSUE THREAD

**Issue by rtbs-dev** [pola-rs/polars#16341]
### Description

I would like to retrieve the intermediary match data from the `regex::Match` struct, e.g. the start, end, range, _and_ string contents. As far as I can tell, this is not available in the polars `.str` accessor, 

Right now, I would have to reverse-engineer the span info by, say, re-running a .find on e.g. substrings of the original, which would introduce a lot of complexity.

**Comment by cmdlineluser** [comment]
Yeah, I think you're right about this not being currently available in `.str.*`

Even a `.str.find_all` would lessen the burden somewhat, but it seems only `.str.find` currently exists.

So this would essentially be `.str.extract_all` but return a list of structs instead?

```python
[ {"start": 5, "end": 7, "match": "..."}, {"start": 9, "end": 13, "match": "..."} ]
```
